driver/Makefile

```diff
@@ -1,8 +1,15 @@
 MPIRUN_CALL ?= mpirun -n 6
 
 test:
-	pytest tests
+	pytest tests/main
 
 test_mpi:
+	MPIRUN_CALL="$(MPIRUN_CALL)" pytest tests/mpi
```

Comment:
This is what I asked for and is I think the best we can do, but if others run into problems and ask about them be aware that `fv3core/tests/mpi` has to be run with `mpirun -n 6 pytest` whereas `driver/tests/mpi` has to be run with just `pytest`. May cause confusion, but should be fine since we have `make test_mpi` targets in both places.